src/fsharp/fsi/fsi.fs

```diff
@@ -1641,18 +1663,124 @@ module internal MagicAssemblyResolution =
                stopProcessingRecovery e range0
                null
 
+#if NETSTANDARD
+        let probingFileNames (name: string) =
+            let isRooted = Path.IsPathRooted name
+            let useSuffix s = not (name.Contains(s + ".") || name.EndsWith(s))      // linux devs often append version # to libraries I.e mydll.so.5.3.2
+            let usePrefix = not (name.Contains(@"\") || name.Contains(@"/"))        // If name has directory information no add no prefix
+            let prefix = [| "lib" |]
+            let suffix = [|
+                    if RuntimeInformation.IsOSPlatform(OSPlatform.Windows) then
+                        ".dll"
+                        ".exe"
+                    elif RuntimeInformation.IsOSPlatform(OSPlatform.OSX) then
```

Comment:
this looks reversed, I'd expect ".so" for linux/*nix and ".dylib" for macos

Reply:
What @smoothdeveloper said.

Reply:
Thanks.